File: src/main/java/com/example/reactive/chapter1/completable/ShoppingCartService.java
package com.example.reactive.chapter1.completable;

import java.util.concurrent.CompletableFuture;

import com.example.reactive.chapter1.commons.Input;
import com.example.reactive.chapter1.commons.Output;

public interface ShoppingCartService {

    CompletableFuture<Output> calculate(Input input);
}

File: src/main/kotlin/wmp/edu/reactive/store/completable/ShoppingCartService.kt
package wmp.edu.reactive.store.completable

import wmp.edu.reactive.store.common.Input
import wmp.edu.reactive.store.common.Output
import java.util.concurrent.CompletableFuture

interface ShoppingCartService {
    fun calculate(input: Input): CompletableFuture<Output>
}
